card/src/main/java/com/adyen/checkout/card/internal/ui/DefaultCardDelegate.kt

```diff
@@ -324,8 +329,8 @@ internal class DefaultCardDelegate(
                 enableLuhnCheck = enableLuhnCheck,
                 isBrandSupported = !shouldFailWithUnsupportedBrand
             ),
-            expiryDateState = validateExpiryDate(inputData.expiryDate, selectedOrFirstCardType?.expiryDatePolicy),
-            securityCodeState = validateSecurityCode(inputData.securityCode, selectedOrFirstCardType),
+            expiryDateState = validateExpiryDate(inputData.expiryDate, reliableSelectedCard?.expiryDatePolicy),
```

Comment:
This change means we won't take the first card into consideration at all if it's not reliable, I'm worried if this will create bugs with the local detection flow. What if we still pass the first detected brand but check inside the validation function whether it's reliable or not?

Reply:
reverted this change as per our discussion offline